SUSE/scf
# bump new autoscaler & consolidate autoscaler instance group & reduce autoscaler parameters in role manifest

## Description
This PR is used to update app-autoscaler release to v1.2.1 , then reduce autoscaler parameters in role manifest by leveraging the sufficient default value in v1.2.1
Also, we combined in app-autoscaler instances into 4 instance groups again, and avoid the hairpin issue by using localhost
Besides this highlighted thing, we also add readiness check and  affinity strategy for app-autoscaler 

## Test plan
1.  install the updated code in vagrant
2.  check all pods running as expected. Note: on my vagrant setup, the api-group pod takes a long time to get ready, then autoscaler-actors and autoscaler-metrics take a long time to get ready as well since they need to connect to Cloud controller first as pre-requisite  before launching all internal processes. 
3. run a small test set:
```
cf api --skip-ssl-validation https://api.cf-dev.io
cf login -u admin -p changeme
cf create-space test
cf target -o system -s test
cf push testapp  -p nodeApp
cf  aasp testapp <policy file>
cf asm testapp memoryused
cf ash testapp
```
4.  ran app-autoscaler-release acceptance test, and passed full test sets.
